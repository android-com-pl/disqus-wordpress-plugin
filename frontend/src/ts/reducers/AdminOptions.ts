import { Record } from 'immutable';

export interface IAdminOptions {
    disqus_forum_url?: string;
    disqus_public_key?: string;
    disqus_secret_key?: string;
    disqus_admin_access_token?: string;
    disqus_installed?: boolean;
    disqus_sso_button?: string;
    disqus_sso_enabled?: boolean;
    disqus_manual_sync?: boolean;
    disqus_sync_token?: string;
    disqus_sync_activated?: boolean;
}

export default class AdminOptions extends Record({
    disqus_admin_access_token: null,
    disqus_forum_url: null,
    disqus_installed: null,
    disqus_manual_sync: null,
    disqus_public_key: null,
    disqus_secret_key: null,
    disqus_sso_button: null,
    disqus_sso_enabled: null,
    disqus_sync_activated: null,
    disqus_sync_token: null,
}) implements IAdminOptions {
    /* tslint:disable:variable-name */
    public disqus_forum_url?: string;
    public disqus_public_key?: string;
    public disqus_secret_key?: string;
    public disqus_admin_access_token?: string;
    public disqus_installed?: boolean;
    public disqus_sso_button?: string;
    public disqus_sso_enabled?: boolean;
    public disqus_manual_sync?: boolean;
    public disqus_sync_activated?: boolean;
    public disqus_sync_token?: string;
    /* tslint:enable:variable-name */

    public constructor(options?: IAdminOptions) {
        super(options);
    }

    public with(values: IAdminOptions) {
        return this.merge(values) as this;
    }
}
