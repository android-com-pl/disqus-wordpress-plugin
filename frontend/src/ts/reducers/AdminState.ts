export interface IMessage {
    text: string;
    type: string;
}

export interface IRestOptions {
    base: string;
    nonce: string;
}

export interface IAdminPermissions {
    canManageSettings: boolean;
}

export interface IAdminUrls {
    disqus: string;
    editComments: string;
    [key: string]: string;
}

export interface IWordpressSite {
    name: string;
}

export interface IAdminOptions {
    disqus_forum_url?: string;
    disqus_public_key?: string;
    disqus_secret_key?: string;
    disqus_admin_access_token?: string;
    disqus_sso_button?: string;
    disqus_sso_enabled?: boolean;
    disqus_manual_sync?: boolean;
    disqus_sync_token?: string;
    [key: string]: any;
}

export interface IAdminConfigData {
    permissions: IAdminPermissions;
    rest: IRestOptions;
    adminUrls: IAdminUrls;
    site: IWordpressSite;
}

export interface IDisqusWordpressWindow extends Window {
    DISQUS_WP: IAdminConfigData;
}

export interface IAdminState {
    config: IAdminConfigData;
    adminOptions: IAdminOptions;
    localAdminOptions: IAdminOptions;
    message: IMessage;
    isSiteFormLocked: boolean;
    [key: string]: any;
}

export default class AdminState implements IAdminState {
    public config: IAdminConfigData;
    public adminOptions: IAdminOptions;
    public localAdminOptions: IAdminOptions;
    public message: IMessage;
    public isSiteFormLocked: boolean;

    constructor(config: IAdminConfigData) {
        this.config = config;
        this.isSiteFormLocked = true;
    }
}
