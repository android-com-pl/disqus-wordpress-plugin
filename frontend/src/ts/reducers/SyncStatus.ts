import { Record } from 'immutable';

export interface ISyncStatus {
    enabled?: boolean;
    subscribed?: boolean;
    subscription?: any;
}

export default class SyncStatus extends Record({
    enabled: null,
    subscribed: null,
    subscription: null,
}) implements ISyncStatus {
    /* tslint:disable:variable-name */
    public enabled?: boolean;
    public subscribed?: boolean;
    public subscription?: any;
    /* tslint:enable:variable-name */

    public constructor(options?: ISyncStatus) {
        super(options);
    }

    public set(key: string, value: any): SyncStatus {
        return super.set(key, value) as this;
    }

    public with(values: ISyncStatus): SyncStatus {
        return this.merge(values) as this;
    }
}
