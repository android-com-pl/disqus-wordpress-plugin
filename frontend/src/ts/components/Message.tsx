import * as React from 'react';
import { IMessage } from '../reducers/AdminState';

const getDismissButton = (props: IMessage) => {
    if (!props.onDismiss)
        return null;

    return (
        <button type="button" className="notice-dismiss" onClick={props.onDismiss}>
            <span className="screen-reader-text">Dismiss this notice.</span>
        </button>
    );
};

const Message = (props: IMessage) => (
    <div className={`notice notice-${props.type} inline is-dismissible`}>
        <p>{props.text}</p>
        {getDismissButton(props)}
    </div>
);

export default Message;
