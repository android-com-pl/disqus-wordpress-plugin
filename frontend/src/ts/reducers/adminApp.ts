import * as Immutable from 'immutable';
import { IAction } from '../actions';
import * as actions from '../actions';
import AdminOptions from './AdminOptions';
import AdminState, { IAdminConfigData, IDisqusWordpressWindow } from './AdminState';

const adminConfigData: IAdminConfigData = (window as IDisqusWordpressWindow).DISQUS_WP;

const initialState: AdminState = new AdminState(adminConfigData);

const adminApp = (state: AdminState = initialState, action: IAction): AdminState => {
    switch (action.type) {
        case actions.UPDATE_ADMIN_OPTIONS:
            state = state.with({
                adminOptions: state.adminOptions.with(action.data),
                localAdminOptions: state.localAdminOptions.with(action.data),
            });
            break;
        case actions.SET_MESSAGE:
            state = state.set('message', action.data);
            break;
        case actions.SET_VALUE:
            state = state.with(action.data);
            break;
        case actions.TOGGLE_VALUE:
            const currentValue: boolean = state.get(action.data, false);
            state = state.with({
                [action.data]: !currentValue,
            });
            break;
        case actions.UPDATE_LOCAL_OPTION:
            state = state.with({
                localAdminOptions: state.localAdminOptions.with(action.data),
            });
            break;
        default:
            break;
    }

    return state;
};

export default adminApp;
