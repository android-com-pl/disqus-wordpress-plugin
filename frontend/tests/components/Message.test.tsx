import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import Message from '../../src/ts/components/Message';
import { IMessage } from '../../src/ts/reducers/AdminState';

describe('Message rendering', () => {
    const props: IMessage = {
        onDismiss: null,
        text: 'Test message',
        type: 'error',
    };

    test('Includes message content and class name', () => {
        const component = TestRenderer.create(<Message {...props} />);
        const root = component.root;

        expect(root.findByType('p').children).toEqual(['Test message']);
        expect(root.findByType('div').props.className).toBe('notice notice-error inline is-dismissible');
    });

    test('Dismiss button is rendered with the onDismiss prop', () => {
        const onDismiss = jest.fn();
        const component = TestRenderer.create(<Message {...props} onDismiss={onDismiss} />);
        const root = component.root;
        const dismissButton = root.findByType('button');
        dismissButton.props.onClick();

        expect(onDismiss).toHaveBeenCalled();
    });

    test('Excludes dismiss button is without the onDismiss prop', () => {
        const component = TestRenderer.create(<Message {...props} />);
        const root = component.root;

        expect(() => {
            root.findByType('button');
        }).toThrow();
    });
});
