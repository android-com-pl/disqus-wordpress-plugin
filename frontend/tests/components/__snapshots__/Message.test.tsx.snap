// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Includes message content and class name 1`] = `
<div
  className="notice notice-error inline is-dismissible"
>
  <p>
    Test message
  </p>
</div>
`;
